shared/lsif/api.ts

```diff
@@ -0,0 +1,91 @@
+import gql from 'tagged-template-noop'
+import { QueryGraphQLFn } from '../util/graphql'
+import { parseGitURI } from '../util/uri'
+
+/** The response envelope for all LSIF queries. */
+export interface GenericLSIFResponse<R> {
+    repository: {
+        commit: {
+            blob: {
+                lsif: R
+            }
+        }
+    }
```

Comment:
Can any of these be `null`?

Reply:
I'm making them nullable to be safe (the previous assumption was that we're _on_ the blob, so it shouldn't be).